### ISSUE TITLE
`apt-get update` fails in 32bit Ubuntu 14.04 container on Ubuntu 16.04 docker host

### ISSUE THREAD

**Issue by ironhouzi** [docker/docker#23059]
**Output of `docker version`:**

```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:43:49 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:43:49 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 4
Server Version: 1.11.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 12
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 4.4.0-22-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.793 GiB
Name: Ubuntu1604Temp
ID: 4VSY:JWJH:TEV6:THCB:N5ZC:KMBU:3HI5:SCXR:IXXD:2E3M:HNP3:CXDO
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

Docker host is a VM

**Steps to reproduce the issue:**
1. `docker run --rm 32bit/ubuntu:14.04 apt-get update`

```
Err http://security.ubuntu.com trusty-security InRelease
  
Err http://extras.ubuntu.com trusty InRelease
  
Err http://security.ubuntu.com trusty-security Release.gpg
  Could not resolve 'security.ubuntu.com'
Err http://extras.ubuntu.com trusty Release.gpg
  Could not resolve 'extras.ubuntu.com'
Err http://archive.ubuntu.com trusty InRelease
  
Err http://archive.ubuntu.com trusty-updates InRelease
  
Err http://archive.ubuntu.com trusty-backports InRelease
  
Err http://archive.ubuntu.com trusty Release.gpg
  Could not resolve 'archive.ubuntu.com'
Err http://archive.ubuntu.com trusty-updates Release.gpg
  Could not resolve 'archive.ubuntu.com'
Err http://archive.ubuntu.com trusty-backports Release.gpg
  Could not resolve 'archive.ubuntu.com'
Reading package lists... Done
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/InRelease  

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/InRelease  

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-backports/InRelease  

W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/trusty-security/InRelease  

W: Failed to fetch http://extras.ubuntu.com/ubuntu/dists/trusty/InRelease  

W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/trusty-security/Release.gpg  Could not resolve 'security.ubuntu.com'

W: Failed to fetch http://extras.ubuntu.com/ubuntu/dists/trusty/Release.gpg  Could not resolve 'extras.ubuntu.com'

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/Release.gpg  Could not resolve 'archive.ubuntu.com'

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/Release.gpg  Could not resolve 'archive.ubuntu.com'

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-backports/Release.gpg  Could not resolve 'archive.ubuntu.com'

W: Some index files failed to download. They have been ignored, or old ones used instead.
```
Expected sucessful apt-get update

The problem doesn't occur when doing `docker run --rm ubuntu apt-get update`, and the problem doesn't occur when doing the same experiement on an ubuntu 14.04 docker host VM.

**Comment by justincormack** [comment]
Which version of `libseccomp` do you have installed? Does it work if you run with `--security-opt seccomp:unconfined`? This looks like possibly the same issue as #22304 as Ubuntu 16.04 ships with a version of libseccomp that is not really compatible with some 32 bit code, although it is kind of odd you see that when running a 14.04 container. Will test on a 16.04 machine.

**Comment by justincormack** [comment]
Yes it works for me if I do `docker run --rm --security-opt seccomp:unconfined 32bit/ubuntu:14.04 apt-get update` so I think it is this issue, will double check.

**Comment by justincormack** [comment]
Ah, interesting, this is actually fixed via #22922. Hmm, I think we better cherry pick this for 1.11.2 then cc @thaJeztah. Hmm, why is this exactly?

**Comment by justincormack** [comment]
Ah I see, the old version of libseccomp that 16.04 shipped (by accident really, as Debian missed the update at the time they shipped) does not really understand the 32 bit `socketcall` stuff properly, so it assumed 32 bit x86 does not uses `socketcall` as it has the new system calls defined, so old code that uses `socketcall` won't work. This probably won't be fixed in 16.04 now I guess, so the fix for ppc64le is also needed for Ubuntu 16.04, and we should add to 1.22.2 (it will be in 1.12 anyway) to fix this.

**Comment by thaJeztah** [comment]
@justincormack looks like this one is resolved, so I'll close this

**Comment by justincormack** [comment]
@ironhouzi this fix will be in the 1.11.2 release.
